django/contrib/contenttypes/fields.py

```diff
@@ -616,9 +648,22 @@ def get_queryset(self):
                 return self._apply_rel_filters(queryset)
 
         def get_prefetch_queryset(self, instances, queryset=None):
+            warnings.warn(
+                "Calling get_prefetch_queryset() is deprecated. "
+                "Use get_prefetch_querysets() instead.",
+                RemovedInDjango60Warning,
+            )
             if queryset is None:
-                queryset = super().get_queryset()
-
+                return self.get_prefetch_querysets(instances)
+            return self.get_prefetch_querysets(instances, [queryset])
+
+        def get_prefetch_querysets(self, instances, querysets=None):
+            if querysets is not None and len(querysets) != 1:
+                raise ValueError(
+                    "querysets argument of get_prefetch_querysets should be of length 1"
+                    f". Received an array of length {len(querysets)}"
```

Comment:
The second sentence seems unnecessary. Also, f-strings should use only plain variable and property access. This guideline is from [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).